bigquery/google/cloud/bigquery/client.py

```diff
@@ -205,6 +205,50 @@ def get_dataset(self, dataset_ref):
             method='GET', path=dataset_ref.path)
         return Dataset.from_api_repr(api_response, self)
 
+    def update_dataset(self, dataset, fields):
+        """Change some fields of a dataset.
+
+        Use ``fields`` to specify which fields to update. At least one field
+        must be provided. If a field is listed in ``fields`` and is ``None`` in
+        ``dataset``, it will be deleted.
+
+        If ``dataset.etag`` is not ``None``, the update will only
+        succeed if the dataset on the server has the same ETag. Thus
+        reading a dataset with ``get_dataset``, changing its fields,
+        and then passing it ``update_dataset`` will ensure that the changes
+        will only be saved if no modifications to the dataset occurred
+        since the read.
+
+        :type dataset: :class:`google.cloud.bigquery.dataset.Dataset`
+        :param dataset: the dataset to update.
+
+        :type fields: sequence of string
+        :param fields: the fields of ``dataset`` to change.
```

Comment:
We should clarify if these fields correspond to the API representation or the way they are represented here in the Python libraries.

Reply:
Done.